components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -268,4 +356,84 @@ private Action updateEndpointAuthenticationProperties(String actionType, String
+            doEndpointAuthenticationValidation(endpointAuthenticationType, endpointAuthentication);
+
+        }
+    }
+
+    /**
+     * Perform pre validations on action model when updating an existing action.
+     * This is specifically used during HTTP PATCH operation and
+     * only validate non-null and non-empty fields.
+     *
+     * @param actionType Action type.
+     * @param action Action update model.
+     * @throws ActionMgtClientException if action model is invalid.
+     */
+    private void doPreUpdateActionValidations(String actionType, Action action) throws ActionMgtClientException {
+
+        if (actionType.equals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType())) {
+            // Validate the action name if present.
+            if (action.getName() != null) {
+                ActionManagementUtil.isValidActionName(action.getName());
+            }
+            // Validate the endpoint URI if present.
+            if (action.getEndpoint() != null && action.getEndpoint().getUri() != null) {
+                ActionManagementUtil.isValidEndpointUri(action.getEndpoint().getUri());
+            }
+            // Validate the endpoint authentication if present.
+            if (action.getEndpoint() != null && action.getEndpoint().getAuthentication() != null) {
+                Authentication endpointAuthentication = action.getEndpoint().getAuthentication();
+                String endpointAuthenticationType = endpointAuthentication.getType().getName();
+                doEndpointAuthenticationValidation(endpointAuthenticationType, endpointAuthentication);
+            }
+        }
+    }
+
+    /**
+     * Perform pre validations on endpoint authentication model.
+     *
+     * @param authenticationType Authentication type.
+     * @param authentication     Endpoint authentication model.
+     * @throws ActionMgtClientException if endpoint authentication model is invalid.
+     */
+    private void doEndpointAuthenticationValidation(String authenticationType, Authentication authentication)
+            throws ActionMgtClientException {
+
+        if (StringUtils.isBlank(authenticationType)) {
+            throw ActionManagementUtil.handleClientException(ActionMgtConstants.ErrorMessages.
+                    ERROR_EMPTY_ENDPOINT_AUTHENTICATION_SCHEME);
+        }
+        if (authenticationType.equals(Authentication.Type.BASIC.getName())) {
+            ActionManagementUtil.isFieldEmpty(authentication.getProperty(Authentication.Property.USERNAME).getValue());
+            ActionManagementUtil.isFieldEmpty(authentication.getProperty(Authentication.Property.PASSWORD).getValue());
+        } else if (authenticationType.equals(Authentication.Type.BEARER.getName())) {
+            String accessToken = authentication.getProperty(Authentication.Property.ACCESS_TOKEN).getValue();
+            ActionManagementUtil.isFieldEmpty(accessToken);
```

Comment:
Can't we directly pass the `authentication.getProperty(Authentication.Property.ACCESS_TOKEN).getValue()` to the method ?